src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -264,6 +275,11 @@ private void CreateObject(ClassDataContract classContract)
                 }
             }
 
+            [DynamicDependency(DynamicallyAccessedMemberTypes.PublicFields, typeof(DBNull))]
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:RequiresUnreferencedCode",
+                Justification = "The DynamicDependency attribute will ensure that DBNull.Value is preserved.")]
+            private static FieldInfo GetDBNullValueField() => Globals.TypeOfDBNull.GetField("Value")!;
```

Comment:
Similar question here - if we didn't use `Globals.TypeOfDBNull` and just used `typeof(DBNull)`, everything would be fine.

Reply:
Same answer here, I'm not familiar enough with the uses of this library to know if it will have an impact to change the caching mechanism. @mconnew what do you think?

Reply:
I believe this is an artifact of this being really old code before some optimizations existed and that there's no negative consequences to using typeof